apis/python/src/tiledbsoma/_common_nd_array.py

```diff
@@ -105,7 +106,8 @@ def shape(self) -> Tuple[int, ...]:
         Lifecycle:
             Experimental.
         """
-        return cast(Tuple[int, ...], tuple(self._soma_reader().shape))
+        handle: SOMAArray = self._handle
```

Comment:
This comment is only tangentially related to this line, so:

Given that `_handle` is now something different for all of the TileDB Array–based types, it’s probably worth it to pull that generic specialization out of `TileDBArray` and put it into each of the concrete ones (so this becomes a `TileDBArray[SOMAArray]`, or however the wrapper type generic stuff works because I kind of forget, etc.).

This isn’t *critical*, so if it ends up being a mess I wouldn’t worry about it but if it is reasonably straightforward it’s worth considering.